cc/blocks/promotion/promotion.js

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+export default async function init(el, promotionPath = '/creativecloud/promotions/hub') {
+  const promotionName = el.getAttribute('data-promotion');
+  const response = await window.fetch(`${promotionPath}/${promotionName}.plain.html`);
+  if (!response.ok) {
+    return;
+  }
+  const promotionContent = await response.text();
+  if (!promotionContent.length) {
+    return;
+  }
+
+  el.appendChild(document.createRange()
```

Comment:
nit: you could just put this in the check 
`if (promotionContent.length) {
    el.appendChild(document.createRange().createContextualFragment(promotionContent));
  }`